spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/mapping/DatastoreMappingContext.java

```diff
@@ -67,15 +67,24 @@ public void setApplicationContext(ApplicationContext applicationContext)
 	/**
 	 * Registers in the DatastoreMappingContext that two classes are discriminated from the
 	 * same Datastore Kind.
-	 * @param classA one class.
-	 * @param classB another class.
+	 * @param parentClass the superclass.
+	 * @param subClass the subclass.
 	 */
-	public static void addDiscriminationClassConnection(Class classA, Class classB) {
-		Set<Class> setA = discriminationFamilies.computeIfAbsent(classA, DatastoreMappingContext::createSingletonSet);
-		Set<Class> setB = discriminationFamilies.computeIfAbsent(classB, DatastoreMappingContext::createSingletonSet);
-
-		setA.addAll(setB);
-		discriminationFamilies.put(classB, setA);
+	public static void addDiscriminationClassConnection(Class parentClass, Class subClass) {
+		Set<Class> setParent = discriminationFamilies.computeIfAbsent(parentClass,
+				unused -> createSingletonSet(subClass));
```

Comment:
if you add subClass later anyway, just return an empty set here

Reply:
Done.